core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -649,7 +649,7 @@ String getFeatureVariableValueForType(@Nonnull String featureKey,
         String variableValue = variable.getDefaultValue();
         Map<String, ?> copiedAttributes = copyAttributes(attributes);
         FeatureDecision featureDecision = decisionService.getVariationForFeature(featureFlag, userId, copiedAttributes);
-        if (featureDecision.variation != null) {
+        if (featureDecision.variation != null && featureDecision.variation.getFeatureEnabled()) {
```

Comment:
Similar feedback as given in Ruby SDK. Here the log message is incorrect for the new scenario, as the user is bucketed into the variation, but because feature is disabled we return default value. Please fix log message corresponding to this scenario.